xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VideoLayerBridgeDRMPRIME.cpp

```diff
@@ -174,6 +200,38 @@ void CVideoLayerBridgeDRMPRIME::Configure(CVideoBufferDRMPRIME* buffer)
     m_DRM->AddProperty(plane, "COLOR_RANGE", value);
 
   auto connector = m_DRM->GetConnector();
+
+  int bpc = 8;
+  bool ycc = false;
+  std::string output_format = "Default";
+
+  if (picture.colorBits > 8)
+  {
+    bpc = 12;
+    ycc = true;
+    output_format = "YCrCb422";
+  }
+
+  std::tie(result, value) = connector->GetPropertyValue("Colorspace", GetColorimetry(picture, ycc));
+  if (result)
+  {
+    CLog::Log(LOGDEBUG, "CVideoLayerBridgeDRMPRIME::{} - setting connector colorspace to {}",
+              __FUNCTION__, GetColorimetry(picture, ycc));
+    m_DRM->AddProperty(connector, "Colorspace", value);
+    m_DRM->SetActive(true);
+  }
+
+  CLog::Log(LOGDEBUG, "CVideoLayerBridgeDRMPRIME::{} - setting max bpc to {} ({})", __FUNCTION__,
+            bpc, result);
+  result = m_DRM->AddProperty(connector, "max bpc", bpc);
+  std::tie(result, value) = connector->GetPropertyValue("output format", output_format);
```

Comment:
Does userspace really care if it's RGB or ycbcr or whatever? How should we know what the right format is?

Reply:
The same question could be asked of the kernel though, with the kernel knowing way less than userspace what the current workload is, and what the user expects.

For your actual question, my playstation will always assume RGB iirc, and fallback to YUV if it can't output it or if you force it in the settings.

Note that this isn't relevant for the latest PR